app/models/pipeline_run.rb

```diff
@@ -967,22 +980,25 @@ def local_amr_drug_summary_path
     "#{LOCAL_AMR_DRUG_SUMMARY_PATH}/#{id}"
   end
 
-  def self.download_file_with_retries(s3_path, destination_dir, max_tries)
+  def self.download_file_with_retries(s3_path, destination, max_tries, dest_is_dir = true)
     round = 0
     while round < max_tries
-      downloaded = PipelineRun.download_file(s3_path, destination_dir)
+      downloaded = PipelineRun.download_file(s3_path, destination, dest_is_dir)
       return downloaded if downloaded
       round += 1
       sleep(15)
     end
   end
 
-  def self.download_file(s3_path, destination_dir)
-    command = "mkdir -p #{destination_dir};"
-    command += "aws s3 cp #{s3_path} #{destination_dir}/;"
-    _stdout, _stderr, status = Open3.capture3(command)
-    return nil unless status.exitstatus.zero?
-    "#{destination_dir}/#{File.basename(s3_path)}"
+  def self.download_file(s3_path, destination, dest_is_dir = true)
```

Comment:
Maybe add a comment "If the destination path is a directory, create the directory if necessary and append the file name to the end of the destination path. If the destination path includes a file name, assume that the parent directories already exist."

Reply:
Took a little while to understand this function.

Reply:
ok, added your comment